### ISSUE TITLE
[Proposal]: Standard way to document tech things in projects

### ISSUE THREAD

**Issue by alessandro-mariotti-zupit** [zupit-it/zupit-standards#23]
### Describe the change

At the moment, technical project documentation, if any, is scattered around various tools (notion, gitlab snippets, markdown files, google drive, etc)
We need some rules for this, and a standard way to document things (and maybe also some suggestions on **what** and **how** to document)
Any suggestion on how to maintain this documentation?
For example:
How do I reach the production VM?
How do I reach that instance of DB? I need to create an SSH tunnel..
What credentials do I use for X, Y, Z? (Obviously, there should be a reference to our password manager here)
My suggestion is to put everything in markdown files in a `docs/` directory in the git repo. For example:
docs/production-release.md
docs/connect-to-vpn.md
docs/i-dont-like-microsoft.md

### Describe how this change might impact positively Zupit

Every project is well documented and (almost) anybody could perform any task on that project

### Describe how this change might impact negatively Zupit

Documenting stuff is boring and requires possibile maintenance

**Comment by alberto-tavoletti-zupit** [comment]
I think we need also a writing standard of the documentation, for example i suggest no aliases: `npm install` instead of `npm i` if it's needed an alias must be specified why and linked to the external docs.
Better be verbose.

I don't know if is preferable to create a specific issue `Documentation writing standard`, let me know.

**Comment by davide-bontempelli-zupit** [comment]
Regarding the tool to use, I am conflicted about whether to keep the documentation in an external tool such as Notion, or in .md files within the project.

Here are the pros of each approach that come to mind:

**External tool pros:**

- Easier to edit and style text and images
- Faster: no need to create a pull request to edit the documentation, which may encourage more contributions
- Possibly more powerful search function
- The product owner (who usually cannot commit to the project) can contribute more easily

**.md files in the project pros:**

- No need to look elsewhere for documentation
- Not dependent on an external service(that we may want to change and that may have a cost )

**Comment by alessio-libardi-zupit** [comment]
What I would do is create the following file structure:

```
.
├── docs/
│   ├── development.md
│   ├── deployments.md
│   └── faq.md
├── CONTRIBUTING.md
└── README.md
```

**Why?**

The `CONTRIBUTING.md` and the `README.md` are a de facto standard for the Open Source Community and the `docs` folder is a de facto standard for the Software Development Community. 

**The case against an external tool**

I would advice against using an external tool because using the Git repository for project documentation offers a centralized, version-controlled, and easy approach that can help ensure our documentation is up-to-date and accessible to all team members and tech contributors.

**Comment by davide-bontempelli-zupit** [comment]
I believe that including a CONTRIBUTING.md file may not be necessary unless the project requires specific guidelines or is open source. In think that in most cases, CONTRIBUTING.md files will tend to have the same content since we are trying to share these guidelines at company-wide level and not per repository

**Comment by alessio-libardi-zupit** [comment]
Yes @davide-bontempelli-zupit you are right! It was more of an example to just do something that is already standard, also the `development.md`, `deployments.md` and `faq.md` were examples.

I don't think there is a way to standardise each and every file name for our projects documentation, and I don't think there should be one. The best we can do is to follow the OS Community using standard file names such as `README` and `CONTRIBUTING` if needed and put everything else in the docs folder.

**Comment by giorgio-betta-zupit** [comment]
A little bit ahead of time but since i'll be proposing Storybook as a tool to manage projects design systems, It would be interesting to take into evaluation the possibility of using Storybook as a container not only for components and design purposes but also for the documentation.

It comes out of the box with a doc page for every story and it ha s useful search functions. It's buildable into a hostable app that can be shared with a client if needed. 
 
[Storybook Docs](https://storybook.js.org/docs/angular/writing-docs/introduction)

**Comment by alessio-libardi-zupit** [comment]
Hi @giorgio-betta-zupit, I think Storybook is a great tool and its docs really shine when documenting components and design systems or style guidelines but I don't think it make much sense to document deployment or debugging resolution strategies such as connecting to the production machine using ssh within Storybook docs.
